cheddar/cheddar-metrics-intercom/src/test/java/com/clicktravel/cheddar/metrics/intercom/IntercomMetricCollectorTest.java

```diff
@@ -0,0 +1,317 @@
+        final Long value = Long.valueOf(randomIntInRange(-10, 10));
+        when(CustomAttribute.newLongAttribute(name, value)).thenReturn(longAttribute);
+        customAttributes.put(name, value);
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        final ArgumentCaptor<CustomAttribute> argumentCaptor = ArgumentCaptor.forClass(CustomAttribute.class);
+        verify(mockUser).addCustomAttribute(argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue(), is(longAttribute));
+        verifyStatic();
+        User.update(mockUser);
+    }
+
+    @Test
+    public void shouldAddFloatCustomAttributeToUser_withFloatAttribute() {
+        // Given
+        final String userId = randomId();
+        final User mockUser = mock(User.class);
+        when(User.find(userId)).thenReturn(mockUser);
+        final Map<String, Object> customAttributes = new HashMap<>();
+        final FloatAttribute floatAttribute = mock(FloatAttribute.class);
+        final String name = randomString(5);
+        final Float value = Float.valueOf(randomIntInRange(-10, 10));
+        when(CustomAttribute.newFloatAttribute(name, value)).thenReturn(floatAttribute);
+        customAttributes.put(name, value);
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        final ArgumentCaptor<CustomAttribute> argumentCaptor = ArgumentCaptor.forClass(CustomAttribute.class);
+        verify(mockUser).addCustomAttribute(argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue(), is(floatAttribute));
+        verifyStatic();
+        User.update(mockUser);
+    }
+
+    @Test
+    public void shouldAddValidCustomAttributesToUser_withValidAndInvalidAttribute() {
+        // Given
+        final String userId = randomId();
+        final User mockUser = mock(User.class);
+        when(User.find(userId)).thenReturn(mockUser);
+        final Map<String, Object> customAttributes = mockCustomAttributes();
+        customAttributes.put(randomString(), new Object());
+
+        // When
+        intercomMetricCollector.addCustomAttributesToUser(userId, customAttributes);
+
+        // Then
+        final ArgumentCaptor<CustomAttribute> argumentCaptor = ArgumentCaptor.forClass(CustomAttribute.class);
+        verify(mockUser, times(classTypes.size())).addCustomAttribute(argumentCaptor.capture());
+        assertThat(argumentCaptor.getAllValues(),
+                containsInAnyOrder(expectedAttributes.toArray(new CustomAttribute[expectedAttributes.size()])));
+        verifyStatic();
+        User.update(mockUser);
+    }
+
+    @Test
+    public void shouldNotAddCustomAttributesToUser_withIntercomFindUserException() {
+        // Given
+        final String userId = randomId();
+        when(User.find(userId)).thenThrow(Exception.class);
+        final Map<String, Object> customAttributes = mockCustomAttributes();
```

Comment:
If we just do `final Map<String, Object> mockCustomAttributes = mock(Map.class);` then this removes the need to have the messy `mockCustomAttributes()` method.  We don't care what the `Map` actually is since it won't be used because of the exception being thrown on the `find` method.